src/System.Net.Security/src/System/Net/_SecureChannel.cs

```diff
@@ -0,0 +1,1487 @@
+                            try
+                            {
+                                WindowsIdentity.RunImpersonated(SafeAccessTokenHandle.InvalidHandle, () =>
+                                {
+                                    store.Open(OpenFlags.ReadOnly | OpenFlags.OpenExistingOnly);
+                                    GlobalLog.Print("SecureChannel::EnsureStoreOpened() storeLocation:" + storeLocation + " returned store:" + store.GetHashCode().ToString("x"));
+                                });
+                            }
+                            catch
+                            {
+                                throw;
+                            }
+
+                            if (isMachineStore)
+                                s_MyMachineCertStoreEx = store;
+                            else
+                                s_MyCertStoreEx = store;
+
+                            return store;
+                        }
+                        catch (Exception exception)
+                        {
+                            if (exception is CryptographicException || exception is SecurityException)
+                            {
+                                GlobalLog.Assert("SecureChannel::EnsureStoreOpened()", "Failed to open cert store, location:" + storeLocation + " exception:" + exception);
+                                return null;
+                            }
+                            if (Logging.On) Logging.PrintError(Logging.Web, SR.Format(SR.net_log_open_store_failed, storeLocation, exception));
+                            throw;
+                        }
+                    }
+                }
+            }
+            return store;
+        }
+
+        private static X509Certificate2 MakeEx(X509Certificate certificate)
+        {
+            Debug.Assert(certificate != null, "certificate != null");
+
+            if (certificate.GetType() == typeof(X509Certificate2))
+            {
+                return (X509Certificate2)certificate;
+            }
+
+            X509Certificate2 certificateEx = null;
+            try
+            {
+                if (certificate.Handle != IntPtr.Zero)
+                {
+                    certificateEx = new X509Certificate2(certificate.Handle);
+                }
+            }
+            catch (SecurityException) { }
+            catch (CryptographicException) { }
+
+            return certificateEx;
+        }
+
+        //
+        // Used only by client SSL code, never returns null.
+        //
+        private string[] GetIssuers()
+        {
+            string[] issuers = Array.Empty<string>();
+
+            if (IsValidContext)
+            {
+                Interop.Secur32.IssuerListInfoEx issuerList = (Interop.Secur32.IssuerListInfoEx)SSPIWrapper.QueryContextAttributes(GlobalSSPI.SSPISecureChannel, _securityContext, Interop.Secur32.ContextAttribute.IssuerListInfoEx);
```

Comment:
I'm not sure what this is doing... is it equivalent to
```C#
X509Chain chain = new X509Chain();
chain.Build(cert);
return chain.ChainElements.Skip(1).Select(ce => ce.Certificate.SubjectName).ToArray();
```

If so, it would be more portable to write it in managed code... though with whatever validation and online/offline policy you wanted.  And with less System.Linq, probably :smile:.